drivers/rtc/rtc_emul.c

```diff
@@ -30,9 +23,9 @@ struct rtc_emul_data {
 
 	struct rtc_time datetime;
 
-	struct k_mutex lock;
+	struct k_spinlock lock;
 
-	struct rtc_emul_work_delayable dwork;
```

Comment:
The mutex->spinlock conversion is agreeably a much more lightweight synchronization primitive.

Just curious - what are we saving by switching from delayable work to timer? It isn't obvious to me if there is a major benefit at this time.

Reply:
1) It's a simpler and IMO more obvious pattern (it removes some lines of code and does not need to be re-scheduled on every interrupt) - that's the main argument IMO
2) It considerably reduces the number of instructions on a time critical path (reducing latency and jitter of the generated tick signal)
3) It reduces the memory footprint of the data structure by 16 bytes IIRC (I looked at it through paholes to verify)

Do you think that justifies the change?

Reply:
Obsolete/removed - see @bjarki-trackunit 's comment below.